### ISSUE TITLE
how to build remarkjs without highlighterjs built-in?

### ISSUE THREAD

**Issue by punkish** [gnab/remark#379]
I am already using highlighterjs in my project, and am already including it in my website. I'd rather continue using that external highlighterjs and build a version of remarkjs that is much smaller and uses the external highlighterjs. How do I do that? `node make` builds a version of remarkjs with the highlighterjs included even if I don't specify the target highlighter

**Comment by tbfe-de** [comment]
I investigated the highlighter-related part of building `remark.js` yesterday and came to the conclusion that as of `remark` version

* v0.13 configuring languages to highlight as described in the Wiki doesn't work any more and
* v0.14 even building the highlighter is completely broken

Note that I'm not a JavaScript expert and may be wrong with the above. Nevertheless I covered it the best I could in issue #381 - you might want to refer the details there.

If my analysis is right - the file `src/remark/src/highlight.js` currently contains **all the highlighting code for many programming languages** resulting in a size of more than 800 KB.

Basically by going back to what v0.12 did and only selecting the programming language C++ I could reduce the size of this file to less than 130 KB ... pretty much an achievement for me but may be in your case still 130 KB too much, as you need no syntax highlighting at all.

I well understand that "just some 100 KB more or less" may not be considered as "relevant" today, but given that `remark` contributes just around 120 KB itself carrying around so much unused code to me seems pretty heavy. (Considering there is no need for any syntax highlighting, as in your case, or just for one or two languageas as in mine.)

**Comment by punkish** [comment]
Thanks for looking into this. I hope we can find a solution for many slideshows don't even have any code. Forcing a ~700KB JS/CSS combo on folks needlessly seems a bit excessive. And for those who are already using highlightjs, or some other highlighter, which is going to be cached already, no point in loading the same thing again. The best way may be to not just provide for building `remarkjs` without `hljs` but also provide a post-slide-load hook that applies the chosen external highlighter to any code in the current slide.

**Comment by gnab** [comment]
It is now possible to [exclude Highlight.js](https://github.com/gnab/remark/wiki/Contributing#excluding-highlighter-from-build) from the build, by bundling an empty highlighter skeleton that doesn't include Highlight.js.

**Comment by punkish** [comment]
Unfortunately not, in my experience. I `git pulled` a fresh version

    ± |develop {1} U:1 ?:2 ✗| → git log
	commit df48281c9ea27ca8ac3d7c249fc641168e671d20
	Author: Ole Petter Bang <olepbang@gmail.com>
	Date:   Tue Oct 25 22:28:25 2016 +0000

		#381: Use explicit Array.sort instead of implicit ShellJS.sort.

	commit 3adfe3d9f0dbbc123d65b7a5e421526814424a1e
	Author: Ole Petter Bang <olepbang@gmail.com>
	Date:   Tue Oct 25 22:00:46 2016 +0000

		#379: Enable excluding highlight.js from build.

Then, I added `"config": { "highlighter": false }`  to **package.json** and ran `node make`. First, I got a test error 

	 |develop {1} U:1 ?:1 ✗| → node make
	Linting...
	Bundling...
	Bundling tests...
	Running tests...
	An error occurred trying to launch phantomjs at "/usr/local/bin/phantomjs": Error: spawn /usr/local/bin/phantomjs ENOENT
	You can specify an explicit path to phantomjs via the `-p` option.
	/Users/punkish/Projects/remark/node_modules/shelljs/src/common.js:360
		  if (config.fatal) throw e;
							^

	Error: exec: internal error
		at Object.error (/Users/punkish/Projects/remark/node_modules/shelljs/src/common.js:71:27)
                …

But a new version of *remarkjs* appeared in the **out** folder nonetheless

	± |develop {1} U:1 ?:2 ✗| → ll out
	total 4576
	drwxr-xr-x   5 punkish  staff      170 Oct 26 08:59 .
	drwxr-xr-x  26 punkish  staff      884 Oct 26 08:59 ..
	-rw-r--r--   1 punkish  staff   936125 Oct 26 08:58 remark.js
	-rw-r--r--   1 punkish  staff  1395599 Oct 26 08:59 tests.js  

936KB remarkjs, certainly not 62KB. What did I do wrong?

**Comment by punkish** [comment]
I take back my previous comment, but with notes: I can now successfully build a slim version of remarkjs through following steps:

    $ node make highlighter
    $ node make lint
    $ node make bundle
    $ node make minify

The above was not clear to me in the notes. I will see if I can improve the build nodes. Many thanks all.
